updater/config.gni

```diff
@@ -3,31 +3,10 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this file,
 # You can obtain one at https://mozilla.org/MPL/2.0/.
 
-declare_args() {
-  # In 2023, Chrome began switching to a new auto-update implementation
-  # ("Omaha 4"). As of this writing, Chrome uses Omaha 4 on macOS and is in the
-  # process of rolling it out on Windows. The new implementation is hidden
-  # behind upstream GN arg `enable_updater`. We set it to true to enable some
-  # integration points. But, as of this writing, we do not yet use Omaha 4. The
-  # arg below enables building our own Omaha 4 integration, which is still
-  # experimental.
-  brave_enable_updater = false
-}
-
 declare_args() {
   # We ship a pre-built (and pre-signed) copy of Omaha 4 in official builds. The
   # main reason for this is that upstream's browser signing code expects it this
   # way. The following arg, which controls whether Omaha 4 should be built from
   # source, is therefore disabled by default for official builds:
-  build_omaha4 = brave_enable_updater && !is_official_build
-}
-
-if (brave_enable_updater) {
-  assert(target_os == "mac",
-         "brave_enable_updater currently only takes effect on macOS.")
-}
-
-if (build_omaha4) {
-  assert(brave_enable_updater,
-         "brave_enable_updater must be true to build Omaha 4")
+  build_omaha4 = !is_official_build
```

Comment:
I'm a bit confused by this, why would we build the updater by default for non-official builds? Also why do we need our own buildflag for this, can't we just use upstream `enable_updater`?

Reply:
> For development. We can change it if you want. But we are building all of the rest of the browser in development builds. Why not do it - and thus test new implementations - for the updater as well?

Wasted build time mostly. Almost no one will actually need this in dev, in particular since it isn't even the default updater yet, so it seems like it should just be enabled manually if you want to build it in dev.

Reply:
or is this only part of create_dist? If it's not part of the normal build them this seems fine. Same with https://github.com/brave/brave-core/pull/25190/files#r1732973998

Reply:
It's part of official builds.

I felt `build_omaha4` wasn't such a great name and renamed it to `use_prebuilt_omaha4` in 043323d6eb9f81a5f9d738616ede9badf7de0d7f. The behavior is still the same though (it builds from source for non-official builds). Let me know if you'd like me to change it to not build from source by default.